src/App.tsx

```diff
@@ -7,18 +7,111 @@ import './App.css'
 import FocusManager from './components/FocusManager'
 
 import AppRoot from './AppRoot'
+import {
+  ScreenReader,
+  AriaScreenReader,
+  SpeechSynthesisTextToSpeech,
+  NullTextToSpeech,
+  TextToSpeech,
+} from './utils/ScreenReader'
 
 /* istanbul ignore next - unsure how to test */
 window.oncontextmenu = (e: MouseEvent): void => {
   e.preventDefault()
 }
 
-const App = () => (
-  <BrowserRouter>
-    <FocusManager>
-      <Route path="/" component={AppRoot} />
-    </FocusManager>
-  </BrowserRouter>
-)
+export interface Props {
+  tts?: {
+    enabled: TextToSpeech
+    disabled: TextToSpeech
+  }
+}
+
+const App = ({
+  tts = {
+    enabled: new SpeechSynthesisTextToSpeech(),
+    disabled: new NullTextToSpeech(),
+  },
+}: Props) => {
+  const [screenReaderEnabled, setScreenReaderEnabled] = useState(false)
+  const [screenReader, setScreenReader] = useState<ScreenReader>(
+    new AriaScreenReader(tts.disabled)
+  )
+
+  /* istanbul ignore next - need to figure out how to test this */
+  const onKeyPress = useCallback(
+    (event: React.KeyboardEvent) => {
+      if (event.key === 'r') {
+        if (screenReaderEnabled) {
+          screenReader.onScreenReaderDisabled()
+          setScreenReader(new AriaScreenReader(tts.disabled))
+          setScreenReaderEnabled(false)
+        } else {
+          const newScreenReader = new AriaScreenReader(tts.enabled)
+          setScreenReader(newScreenReader)
+          setScreenReaderEnabled(true)
+          newScreenReader.onScreenReaderEnabled()
+        }
+      }
+    },
+    [
+      screenReader,
+      setScreenReader,
+      screenReaderEnabled,
+      setScreenReaderEnabled,
+      tts.disabled,
+      tts.enabled,
+    ]
+  )
+
+  /* istanbul ignore next - need to figure out how to test this */
+  const onClick = useCallback(
+    ({ target }: React.MouseEvent) => {
+      if (target) {
+        const currentPath = window.location.pathname
+
+        setImmediate(() => {
```

Comment:
Is it safe to use `setImmediate`? 

- https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate
- https://caniuse.com/#feat=setimmediate

Perhaps we need to add `setImmediate` to our `package.json` dependencies?

Reply:
That’s interesting. It does work, so where is it coming from…?